src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/auth/signin/DetachedSignInPage.java

```diff
@@ -0,0 +1,37 @@
+package com.wikia.webdriver.pageobjectsfactory.pageobject.auth.signin;
+
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.forgotpassword.DetachedForgotPasswordPage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.forgotpassword.ForgotPasswordPage;
+
+public class DetachedSignInPage extends BasePageObject implements SignInPage {
+
+  private SignInPage signInPage;
+
+  public DetachedSignInPage() {
+    this.signInPage = new AttachedSignInPage();
+  }
+
+  public DetachedSignInPage(SignInPage signInPage) {
+    this.signInPage = signInPage;
+  }
+
+  @Override public ForgotPasswordPage clickForgotPasswordLink() {
+    ForgotPasswordPage forgotPassword = this.signInPage.clickForgotPasswordLink();
+    return new DetachedForgotPasswordPage(forgotPassword);
+  }
+
+  @Override public SignInPage typePassword(String password) {
+    this.signInPage.typePassword(password);
+    return this;
+  }
+
+  @Override public void login(String username, String password) {
+
+  }
+
+  @Override public void login(User user) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1186)